sunbeam-python/sunbeam/commands/launch.py

```diff
@@ -60,10 +60,21 @@ def launch(
     data_location = snap.paths.user_data
     deployment: Deployment = ctx.obj
     jhelper = JujuHelper(deployment.get_connected_controller())
+
+    is_bootstrapped = deployment.get_client().cluster.check_sunbeam_bootstrapped()
```

Comment:
Can this be moved before `get_connected_controller`, as per my previous comment?

It's cheaper to fail because Sunbeam is not bootstrapped than because we don't have a controller yet :)